File: frontend/src/types/tradingPartner.ts
export interface TradingPartner {
  id: string;
  companyName: string;
  companyCode: string;
  partnerType: string;
  contactPerson?: string;
  contactEmail?: string;
  contactPhone?: string;
  address?: string;
  taxNumber?: string;
  
  // Credit Management
  creditLimit: number;
  creditLimitValidUntil: string;
  paymentTermDays: number;
  currentExposure: number;
  creditUtilization: number;
  
  // Statistics
  totalPurchaseAmount: number;
  totalSalesAmount: number;
  totalTransactions: number;
  lastTransactionDate?: string;
  
  // Status
  isActive: boolean;
  isBlocked: boolean;
  blockReason?: string;
  
  // Calculated fields
  isCreditExceeded: boolean;
  isCreditExpired: boolean;
}

export interface TradingPartnerSummary {
  id: string;
  companyName: string;
  companyCode: string;
  partnerType: string;

  // Contact Information
  contactPerson?: string;
  contactEmail?: string;
  contactPhone?: string;
  address?: string;
  taxNumber?: string;

  // Credit Management - MUST PERSIST
  creditLimit: number;
  creditLimitValidUntil: string;  // ← Added to fix persistence
  paymentTermDays: number;  // ← Added to fix persistence
  currentExposure: number;
  creditUtilization: number;

  // Status - MUST PERSIST
  isActive: boolean;
  isBlocked: boolean;  // ← Added to fix persistence
  blockReason?: string;  // ← Added to fix persistence
  isCreditExceeded: boolean;
}

export interface CreateTradingPartnerRequest {
  companyName: string;
  partnerType: string;
  contactPerson?: string;
  contactEmail?: string;
  contactPhone?: string;
  address?: string;
  taxNumber?: string;
  creditLimit: number;
  creditLimitValidUntil: string;
  paymentTermDays?: number;
  isActive?: boolean;
  isBlocked?: boolean;
  blockReason?: string;
}

export interface UpdateTradingPartnerRequest {
  companyName: string;
  contactPerson?: string;
  contactEmail?: string;
  contactPhone?: string;
  address?: string;
  taxNumber?: string;
  creditLimit: number;
  creditLimitValidUntil: string;
  paymentTermDays: number;
  isActive: boolean;
  isBlocked: boolean;
  blockReason?: string;
}

export interface GetTradingPartnersParams {
  partnerType?: string;
  isActive?: boolean;
  searchTerm?: string;
}

export interface PagedResult<T> {
  data: T[];
  page: number;
  pageSize: number;
  totalItems: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}

export enum PartnerType {
  Supplier = 'Supplier',
  Customer = 'Customer',
  Both = 'Both'
}

// Import financial report related types
import { TradingPartnerAnalysis, FinancialReportSummary } from './financialReport';

// Extended interfaces for analysis
export interface TradingPartnerWithAnalysis extends TradingPartner {
  analysis?: TradingPartnerAnalysis;
  financialReports?: FinancialReportSummary[];
}

File: src/OilTrading.Core/Entities/TradingPartner.cs
using OilTrading.Core.Common;

namespace OilTrading.Core.Entities;

public class TradingPartner : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Code { get; set; } = string.Empty;
    public string CompanyName { get; set; } = string.Empty;
    public string CompanyCode { get; set; } = string.Empty;
    public TradingPartnerType Type { get; set; }
    public TradingPartnerType PartnerType { get; set; }
    public string? ContactPerson { get; set; }
    public string ContactEmail { get; set; } = string.Empty;
    public string ContactPhone { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string Country { get; set; } = string.Empty;
    public string TaxId { get; set; } = string.Empty;
    public string? TaxNumber { get; set; }
    public string ContactInfo { get; set; } = string.Empty;
    
    // Credit Management
    public decimal CreditLimit { get; set; }
    public DateTime CreditLimitValidUntil { get; set; }
    public int PaymentTermDays { get; set; } = 30;
    public decimal CurrentExposure { get; set; }
    public string CreditRating { get; set; } = string.Empty;
    
    // Statistics
    public decimal TotalPurchaseAmount { get; set; }
    public decimal TotalSalesAmount { get; set; }
    public int TotalTransactions { get; set; }
    public DateTime? LastTransactionDate { get; set; }
    
    // Status
    public bool IsActive { get; set; } = true;
    public bool IsBlocked { get; set; } = false;
    public string? BlockReason { get; set; }
    
    // Navigation Properties
    public ICollection<PurchaseContract> PurchaseContracts { get; set; } = [];
    public ICollection<SalesContract> SalesContracts { get; set; } = [];
    public ICollection<PhysicalContract> PhysicalContracts { get; set; } = new List<PhysicalContract>();
    public ICollection<FinancialReport> FinancialReports { get; set; } = [];
    public ICollection<PaymentRiskAlert> PaymentRiskAlerts { get; set; } = [];
}

public enum TradingPartnerType
{
    Supplier = 1,
    Customer = 2,
    Both = 3,
    Trader = 4,
    EndUser = 5
}